packages/macro/src/macroJsx.ts

```diff
@@ -51,6 +51,21 @@ export default class MacroJSX {
     this.elementIndex = makeCounter()
   }
 
+  safeJsxAttribute = (name: string, value: string) => {
+    // Quoted JSX attributes use XML-style escapes instead of JavaScript-style escapes.
+    // This means that <Trans id="Say \"hi\"!" /> is invalid, but <Trans id={"Say \"hi\"!"} /> is valid.
+
+    // We could consider removing this condition and always wrap in a jsxExpressionContainer.
```

Comment:
Removing the conditional here seems safe – I've done this in a larger codebase. 

Let me know if we should remove the conditional here (a lot of the tests would need to be updated).

Reply:
@semoal Let me know if we should remove the conditional here (a lot of the tests would need to be updated).

Reply:
Will that imply a breaking change? Sorry but I don't get why we should update the tests?

Reply:
It will not introduce a breaking change, but as all id and and message attribute will now be wrapped in an expression, then all the tests would need to be updated.

So instead of `<Trans id="test" />` it would be `<Trans id={"test"} />`. Again not breaking change.

Keeping the conditional might be a fine tradeoff, then we only wrap the attributes in an expression in case there is a `"` in them.

Reply:
We should remove the conditional here to handle additional corner cases. This was done in https://github.com/lingui/js-lingui/pull/1234